components/brave_wallet/browser/zcash/zcash_keyring.cc

```diff
@@ -44,6 +65,97 @@ std::string ZCashKeyring::GetAddressInternal(const HDKey& hd_key) const {
   return hd_key.GetZCashTransparentAddress(testnet_);
 }
 
+std::optional<std::vector<uint8_t>> ZCashKeyring::GetPubkeyHash(
+    const mojom::ZCashKeyId& key_id) {
+  auto hd_key_base = DeriveKey(key_id);
+  if (!hd_key_base) {
+    return std::nullopt;
+  }
+
+  return Hash160(hd_key_base->GetPublicKeyBytes());
+}
+
+#if BUILDFLAG(ENABLE_ORCHARD)
+std::optional<std::string> ZCashKeyring::GetUnifiedAddress(
+    const mojom::ZCashKeyId& transparent_key_id,
+    const mojom::ZCashKeyId& orchard_key_id) {
+  if (!orchard_key_) {
+    return std::nullopt;
```

Comment:
providing no information about why these failed seems like it will make debugging very difficult. Can we use `base::expected` here instead? This comes back to https://github.com/brave/brave-core/pull/22870/files/04eb59028e8c03b6c16a5457d53f55564a7cf113#r1575228107 because it seems like we're not in a valid state here in the first place

Reply:
I am not sure if this is a right place to do because in other keyrings we use optional, for ex. bitcoin keyring. Maybe @supermassive can consider on switching to expected everywhere since he is now doing some refactoring.

Reply:
https://github.com/brave/brave-core/pull/22870/files#r1575289868

Reply:
Let's keep it as is now. And refactor later so keryings have more predictable state in that sense

Reply:
can you create an issue for it?

Reply:
https://github.com/brave/brave-browser/issues/37856